src/NLog/Targets/TargetWithContext.cs

```diff
@@ -276,6 +277,175 @@ protected bool ShouldIncludeProperties(LogEventInfo logEvent)
             return combinedProperties ?? new Dictionary<string, object>(StringComparer.Ordinal);
         }
 
+        private sealed class GetAllPropertiesCountContext
+        {
+            private readonly TargetWithContext _targetContext;
+            private readonly LogEventInfo _logEvent;
+
+            private IDictionary<object, object> _logEventProperties;
+            private IEnumerable<KeyValuePair<string, object>> _scopeContextProperties;
+            private ICollection<string> _gdcNames;
+
+            private bool _CountPropertiesCalled;
+            private int _propertiesCount;
+            private bool _checkExcludeProperties;
+
+            private KeyValuePair<string, object>? BuildPair(string propertyName, object propertyValue)
+            {
+                if (string.IsNullOrEmpty(propertyName))
+                    return null;
+
+                if (_checkExcludeProperties && _targetContext.ExcludeProperties.Contains(propertyName))
+                    return null;
+
+                return new KeyValuePair<string, object>(propertyName, propertyValue);
+            }
+
+            private KeyValuePair<string, object>? BuildPairSerialized(string propertyName, object propertyValue)
+            {
+                if (string.IsNullOrEmpty(propertyName))
+                    return null;
+
+                if (_checkExcludeProperties && _targetContext.ExcludeProperties.Contains(propertyName))
+                    return null;
+
+                if (!_targetContext.SerializeScopeContextProperty(_logEvent, propertyName, propertyValue, out var serializedValue))
+                    return null;
+
+                return new KeyValuePair<string, object>(propertyName, serializedValue);
+            }
+
+            public int PropertiesCount
+            {
+                get => _propertiesCount;
+            }
+
+            private GetAllPropertiesCountContext() { }
```

Comment:
Any reason for the private constructor ?

Reply:
Extra caution means ensuring that required objects (TargetWithContext  &  LogEventInfo) are passed to the constructor.

After second thought, this is an internal class, not a part of any API. Really "**extra caution**".

Deleted.